nad_ch/domain/entities.py

```diff
@@ -55,3 +57,6 @@ def generate_filename(file_path: str, provider: DataProvider) -> str:
         _, file_extension = os.path.splitext(file_path)
         filename = f"{formatted_provider_name}_{datetime_str}{file_extension}"
         return filename
+
+    def has_report(self) -> bool:
+        return False if self.report is None or not self.report else True
```

Comment:
Do you need truthiness here? Otherwise, I'd be more direct.

```suggestion
        return self.report is not None
```